FakebookNotifications/FakebookNotifications.WebApi/Startup.cs

```diff
@@ -26,6 +29,17 @@ public Startup(IConfiguration configuration)
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
+            //MongoDb service Config
+            services.Configure<NotificationsDatabaseSettings>(
+                Configuration.GetSection(nameof(NotificationsDatabaseSettings)));
+
+            //Mongo add db service
+            services.AddSingleton<INotificationsDatabaseSettings>(sp =>
+                sp.GetRequiredService<IOptions<NotificationsDatabaseSettings>>().Value);
```

Comment:
i'd add package reference Microsoft.Extensions.Options to the data access layer and then just request IOptions<NotificationsDatabaseSettings> where you wind up needing it, rather than reregister the service. IOptions is a fine dependency to take, it's pretty generic.

the comment is wrong though, this doesnt add any db service, just some settings.
drop the comments below too